my-app/src/app/app-routing.module.ts

```diff
@@ -1,15 +1,16 @@
 import { NgModule } from '@angular/core';
 import { Routes, RouterModule } from '@angular/router';
+import { AuthGuard } from './core/guards/auth.guard';
+import { LoginComponent } from './login/login.component';
 import { MainLayoutComponent } from './main-layout/main-layout.component';
 
 // Task 2: Basic redirect route
 const routes: Routes = [
+  { path: 'courses', component: MainLayoutComponent, canActivate: [AuthGuard] },
```

Comment:
I think some corrections are required for auth-guard. I tried to do this via routing, but it was not working properly.